packages/components/src/components/button/button.e2e-test.ts

```diff
@@ -1,66 +1,554 @@
-import { test } from '../../../config/playwright/setup';
+/* eslint-disable no-await-in-loop */
+/* eslint-disable no-restricted-syntax */
+import { expect } from '@playwright/test';
+import { ComponentsPage, test } from '../../../config/playwright/setup';
+import StickerSheet from '../../../config/playwright/setup/utils/Stickersheet';
+import { BUTTON_COLORS, BUTTON_VARIANTS, DEFAULTS, ICON_BUTTON_SIZES, PILL_BUTTON_SIZES } from './button.constants';
 
-test.beforeEach(async ({ componentsPage }) => {
-  await componentsPage.mount({
-    html: `
-        <mdc-button />
-      `,
+type SetupOptions = {
+  componentsPage: ComponentsPage;
+  prefixIcon?: string;
+  postfixIcon?: string;
+  active?: boolean;
+  disabled?: boolean;
+  softDisabled?: boolean;
+  variant?: string;
+  size?: number;
+  color?: string;
+  children?: any;
+  ariaLabel?: string;
+  secondButtonForFocus?: boolean;
+};
+
+const setup = async (args: SetupOptions) => {
+  const { componentsPage, ...restArgs } = args;
+  if (restArgs.children) {
+    await componentsPage.mount({
+      html: `
+      ${restArgs.secondButtonForFocus ? '<div id="wrapper">' : ''}
+        <mdc-button
+          ${restArgs.active ? 'active' : ''}
+          ${restArgs.disabled ? 'disabled' : ''}
+          ${restArgs.softDisabled ? 'soft-disabled' : ''}
+          ${restArgs.variant ? `variant="${restArgs.variant}"` : ''}
+          ${restArgs.size ? `size="${restArgs.size}"` : ''}
+          ${restArgs.color ? `color="${restArgs.color}"` : ''}
+          ${restArgs.prefixIcon ? `prefix-icon="${restArgs.prefixIcon}"` : ''}
+          ${restArgs.postfixIcon ? `postfix-icon="${restArgs.postfixIcon}"` : ''}
+        >
+          ${restArgs.children}
+        </mdc-button>
+      ${restArgs.secondButtonForFocus ? '<mdc-button>Second Button</mdc-button></div>' : ''}
+        `,
+      clearDocument: true,
+    });
+  } else {
+    await componentsPage.mount({
+      html: `
+      ${restArgs.secondButtonForFocus ? '<div id="wrapper">' : ''}
+        <mdc-button
+          ${restArgs.active ? 'active' : ''}
+          ${restArgs.disabled ? 'disabled' : ''}
+          ${restArgs.softDisabled ? 'soft-disabled' : ''}
+          ${restArgs.variant ? `variant="${restArgs.variant}"` : ''}
+          ${restArgs.size ? `size="${restArgs.size}"` : ''}
+          ${restArgs.color ? `color="${restArgs.color}"` : ''}
+          ${restArgs.prefixIcon ? `prefix-icon="${restArgs.prefixIcon}"` : ''}
+          ${restArgs.postfixIcon ? `postfix-icon="${restArgs.postfixIcon}"` : ''}
+          ${restArgs.ariaLabel ? `aria-label="${restArgs.ariaLabel}"` : ''}
+        ></mdc-button>
+      ${restArgs.secondButtonForFocus ? '<mdc-button>Second Button</mdc-button></div>' : ''}
+        `,
+      clearDocument: true,
+    });
+  }
```

Comment:
[suggestion]: This can reduced like this
```suggestion

  const htmlButtonContent = `
    ${restArgs.secondButtonForFocus ? '<div id="wrapper">' : ''}
      <mdc-button
        ${restArgs.active ? 'active' : ''}
        ${restArgs.disabled ? 'disabled' : ''}
        ${restArgs.softDisabled ? 'soft-disabled' : ''}
        ${restArgs.variant ? `variant="${restArgs.variant}"` : ''}
        ${restArgs.size ? `size="${restArgs.size}"` : ''}
        ${restArgs.color ? `color="${restArgs.color}"` : ''}
        ${restArgs.prefixIcon ? `prefix-icon="${restArgs.prefixIcon}"` : ''}
        ${restArgs.postfixIcon ? `postfix-icon="${restArgs.postfixIcon}"` : ''}
        ${restArgs.ariaLabel ? `aria-label="${restArgs.ariaLabel}"` : ''}
      >
        ${restArgs.children ? restArgs.children : ''}
      </mdc-button>
    ${restArgs.secondButtonForFocus ? '<mdc-button>Second Button</mdc-button></div>' : ''}
  `;
  await componentsPage.mount({
    html: htmlButtonContent,
    clearDocument: true,
  });
```

Reply:
Initially, I had created it in this way. However, `${restArgs.children ? restArgs.children : ''}` adds an empty value between opening and closing tag of the button, which is considered in the slot value. Thus, I had to make it verbose.

We are determining the type of the button based on the children values that are passed into the slot. For icon buttons (rounded in shape) we shouldn't have any children/slot values.